core/transaction_pool_test.go

```diff
@@ -97,18 +97,18 @@ func TestTransactionQueue(t *testing.T) {
 	tx1.SignECDSA(key)
 	tx2.SignECDSA(key)
 	tx3.SignECDSA(key)
-	pool.queueTx(tx1)
-	pool.queueTx(tx2)
-	pool.queueTx(tx3)
+	pool.queueTx(tx1.Hash(), tx1)
+	pool.queueTx(tx2.Hash(), tx2)
+	pool.queueTx(tx3.Hash(), tx3)
 	from, _ = tx1.From()
+
 	pool.checkQueue()
 
 	if len(pool.txs) != 1 {
 		t.Error("expected tx pool to be 1 =")
 	}
-
-	if len(pool.queue[from]) != 3 {
-		t.Error("expected transaction queue to be empty. is", len(pool.queue[from]))
+	if len(pool.queue[from]) != 2 {
```

Comment:
About this test: My understanding of the queue is that it should contain transactions that cannot be processed yet.
During `checkQueue`, all processable transactions are moved to the main pool.

The test started failing when I added a `delete` to actually remove txs from the queue when they get moved to `pool.txs`. I believe that the correct count is indeed 2 because the first tx (with nonce 0) is processable and moved to
`pool.txs`. The two remaining transactions are not moved and stay in the pool.

```
before checkQueue: (*core.TxPool)(0xc2080bb590)({
 quit: (chan bool) 0xc20805f560,
 currentState: (core.stateFn) 0x409b1c0,
 gasLimit: (func() *big.Int) 0x409b1d0,
 eventMux: (*event.TypeMux)(0xc2080b0fc0)({
  mutex: (sync.RWMutex) {
   w: (sync.Mutex) {
    state: (int32) 0,
    sema: (uint32) 0
   },
   writerSem: (uint32) 0,
   readerSem: (uint32) 0,
   readerCount: (int32) 0,
   readerWait: (int32) 0
  },
  subm: (map[reflect.Type][]*event.muxsub) <nil>,
  stopped: (bool) false
 }),
 mu: (sync.RWMutex) {
  w: (sync.Mutex) {
   state: (int32) 0,
   sema: (uint32) 0
  },
  writerSem: (uint32) 0,
  readerSem: (uint32) 0,
  readerCount: (int32) 0,
  readerWait: (int32) 0
 },
 txs: (map[common.Hash]*types.Transaction) {
 },
 queue: (map[common.Address]map[common.Hash]*types.Transaction) (len=1) {
  (common.Address) (len=20 cap=20) {
   00000000  23 f3 92 15 21 b0 d8 15  0f 70 77 eb ff 51 37 48  |#...!....pw..Q7H|
   00000010  ed 43 10 c9                                       |.C..|
  }: (map[common.Hash]*types.Transaction) (len=3) {
   (common.Hash) (len=32 cap=32) {
    00000000  60 d6 04 7a 24 6e 33 27  51 2d dd 90 d6 37 47 c1  |`..z$n3'Q-...7G.|
    00000010  07 de c9 95 aa bf a5 b0  27 4a 5b 52 2c f6 d7 5e  |........'J[R,..^|
   }: (*types.Transaction)(0xc20805f5c0)(
	TX(60d6047a246e3327512ddd90d63747c107dec995aabfa5b0274a5b522cf6d75e)
	Contract: false
	From:     23f3921521b0d8150f7077ebff513748ed4310c9
	To:       0000000000000000000000000000000000000000
	Nonce:    0
	GasPrice: 100
	GasLimit  100
	Value:    100
	Data:     0x
	V:        0x1b
	R:        0xd9c39e19c5e34157dc60a3a92e21338e766439a2d67b320220c96fc358d55739
	S:        0x4131f4e97b3773f2024193e07bfbea360094932fd800ceb2ef99192a475767f0
	Hex:      f85d80646494000000000000000000000000000000000000000064801ba0d9c39e19c5e34157dc60a3a92e21338e766439a2d67b320220c96fc358d55739a04131f4e97b3773f2024193e07bfbea360094932fd800ceb2ef99192a475767f0
),
   (common.Hash) (len=32 cap=32) {
    00000000  84 78 d6 97 79 34 55 9e  2c 61 90 25 96 9a b2 83  |.x..y4U.,a.%....|
    00000010  d5 e3 92 44 eb e1 e1 7d  5b 86 1f 45 c9 4b d3 b1  |...D...}[..E.K..|
   }: (*types.Transaction)(0xc20805f620)(
	TX(8478d6977934559e2c619025969ab283d5e39244ebe1e17d5b861f45c94bd3b1)
	Contract: false
	From:     23f3921521b0d8150f7077ebff513748ed4310c9
	To:       0000000000000000000000000000000000000000
	Nonce:    10
	GasPrice: 100
	GasLimit  100
	Value:    100
	Data:     0x
	V:        0x1c
	R:        0x66970067d485119301ed48acea4dc47415d7a9e1cf2e451424aa22d6e9a301f6
	S:        0x54be61caa1df50da3a7f84d317c4a4c9633458d5831c1d6f8d090ffc7bfbe03c
	Hex:      f85d0a646494000000000000000000000000000000000000000064801ca066970067d485119301ed48acea4dc47415d7a9e1cf2e451424aa22d6e9a301f6a054be61caa1df50da3a7f84d317c4a4c9633458d5831c1d6f8d090ffc7bfbe03c
),
   (common.Hash) (len=32 cap=32) {
    00000000  9b 30 20 7a c7 99 c7 a1  ff eb 28 61 06 09 4c e0  |.0 z......(a..L.|
    00000010  2a 79 6c d2 e5 76 f5 b6  e3 d7 0b 09 85 7f c6 1c  |*yl..v..........|
   }: (*types.Transaction)(0xc20805f680)(
	TX(9b30207ac799c7a1ffeb286106094ce02a796cd2e576f5b6e3d70b09857fc61c)
	Contract: false
	From:     23f3921521b0d8150f7077ebff513748ed4310c9
	To:       0000000000000000000000000000000000000000
	Nonce:    11
	GasPrice: 100
	GasLimit  100
	Value:    100
	Data:     0x
	V:        0x1b
	R:        0x6a8227b6004e83bc79cbbf9cf7b05d3a5f5e7fa74124db50e24bb4367194b593
	S:        0x208f203bec47ac8cc540b20dd56a23f7ca2996236330977c3e09c084c3bcfcf8
	Hex:      f85d0b646494000000000000000000000000000000000000000064801ba06a8227b6004e83bc79cbbf9cf7b05d3a5f5e7fa74124db50e24bb4367194b593a0208f203bec47ac8cc540b20dd56a23f7ca2996236330977c3e09c084c3bcfcf8
)
  }
 }
})

after checkQueue: (*core.TxPool)(0xc2080bb590)({
 quit: (chan bool) 0xc20805f560,
 currentState: (core.stateFn) 0x409b1c0,
 gasLimit: (func() *big.Int) 0x409b1d0,
 eventMux: (*event.TypeMux)(0xc2080b0fc0)({
  mutex: (sync.RWMutex) {
   w: (sync.Mutex) {
    state: (int32) 0,
    sema: (uint32) 0
   },
   writerSem: (uint32) 0,
   readerSem: (uint32) 0,
   readerCount: (int32) 0,
   readerWait: (int32) 0
  },
  subm: (map[reflect.Type][]*event.muxsub) <nil>,
  stopped: (bool) false
 }),
 mu: (sync.RWMutex) {
  w: (sync.Mutex) {
   state: (int32) 0,
   sema: (uint32) 0
  },
  writerSem: (uint32) 0,
  readerSem: (uint32) 0,
  readerCount: (int32) 0,
  readerWait: (int32) 0
 },
 txs: (map[common.Hash]*types.Transaction) (len=1) {
  (common.Hash) (len=32 cap=32) {
   00000000  60 d6 04 7a 24 6e 33 27  51 2d dd 90 d6 37 47 c1  |`..z$n3'Q-...7G.|
   00000010  07 de c9 95 aa bf a5 b0  27 4a 5b 52 2c f6 d7 5e  |........'J[R,..^|
  }: (*types.Transaction)(0xc20805f5c0)(
	TX(60d6047a246e3327512ddd90d63747c107dec995aabfa5b0274a5b522cf6d75e)
	Contract: false
	From:     23f3921521b0d8150f7077ebff513748ed4310c9
	To:       0000000000000000000000000000000000000000
	Nonce:    0
	GasPrice: 100
	GasLimit  100
	Value:    100
	Data:     0x
	V:        0x1b
	R:        0xd9c39e19c5e34157dc60a3a92e21338e766439a2d67b320220c96fc358d55739
	S:        0x4131f4e97b3773f2024193e07bfbea360094932fd800ceb2ef99192a475767f0
	Hex:      f85d80646494000000000000000000000000000000000000000064801ba0d9c39e19c5e34157dc60a3a92e21338e766439a2d67b320220c96fc358d55739a04131f4e97b3773f2024193e07bfbea360094932fd800ceb2ef99192a475767f0
)
 },
 queue: (map[common.Address]map[common.Hash]*types.Transaction) (len=1) {
  (common.Address) (len=20 cap=20) {
   00000000  23 f3 92 15 21 b0 d8 15  0f 70 77 eb ff 51 37 48  |#...!....pw..Q7H|
   00000010  ed 43 10 c9                                       |.C..|
  }: (map[common.Hash]*types.Transaction) (len=2) {
   (common.Hash) (len=32 cap=32) {
    00000000  84 78 d6 97 79 34 55 9e  2c 61 90 25 96 9a b2 83  |.x..y4U.,a.%....|
    00000010  d5 e3 92 44 eb e1 e1 7d  5b 86 1f 45 c9 4b d3 b1  |...D...}[..E.K..|
   }: (*types.Transaction)(0xc20805f620)(
	TX(8478d6977934559e2c619025969ab283d5e39244ebe1e17d5b861f45c94bd3b1)
	Contract: false
	From:     23f3921521b0d8150f7077ebff513748ed4310c9
	To:       0000000000000000000000000000000000000000
	Nonce:    10
	GasPrice: 100
	GasLimit  100
	Value:    100
	Data:     0x
	V:        0x1c
	R:        0x66970067d485119301ed48acea4dc47415d7a9e1cf2e451424aa22d6e9a301f6
	S:        0x54be61caa1df50da3a7f84d317c4a4c9633458d5831c1d6f8d090ffc7bfbe03c
	Hex:      f85d0a646494000000000000000000000000000000000000000064801ca066970067d485119301ed48acea4dc47415d7a9e1cf2e451424aa22d6e9a301f6a054be61caa1df50da3a7f84d317c4a4c9633458d5831c1d6f8d090ffc7bfbe03c
),
   (common.Hash) (len=32 cap=32) {
    00000000  9b 30 20 7a c7 99 c7 a1  ff eb 28 61 06 09 4c e0  |.0 z......(a..L.|
    00000010  2a 79 6c d2 e5 76 f5 b6  e3 d7 0b 09 85 7f c6 1c  |*yl..v..........|
   }: (*types.Transaction)(0xc20805f680)(
	TX(9b30207ac799c7a1ffeb286106094ce02a796cd2e576f5b6e3d70b09857fc61c)
	Contract: false
	From:     23f3921521b0d8150f7077ebff513748ed4310c9
	To:       0000000000000000000000000000000000000000
	Nonce:    11
	GasPrice: 100
	GasLimit  100
	Value:    100
	Data:     0x
	V:        0x1b
	R:        0x6a8227b6004e83bc79cbbf9cf7b05d3a5f5e7fa74124db50e24bb4367194b593
	S:        0x208f203bec47ac8cc540b20dd56a23f7ca2996236330977c3e09c084c3bcfcf8
	Hex:      f85d0b646494000000000000000000000000000000000000000064801ba06a8227b6004e83bc79cbbf9cf7b05d3a5f5e7fa74124db50e24bb4367194b593a0208f203bec47ac8cc540b20dd56a23f7ca2996236330977c3e09c084c3bcfcf8
)
  }
 }
})
```